Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ExpressionFilterFactory.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Inspections.Concrete.UnreachableCaseInspection{
+
+    public static class ExpressionFilterFactory
+    {
+        //The following MIN/MAX values relate to VBA types
+        private static class CompareExtents
+        {
+            public static long LONGMIN = Int32.MinValue; //- 2147486648;
+            public static long LONGMAX = Int32.MaxValue; //2147486647
+            public static long INTEGERMIN = Int16.MinValue; //- 32768;
+            public static long INTEGERMAX = Int16.MaxValue; //32767
+            public static long BYTEMIN = byte.MinValue;  //0
+            public static long BYTEMAX = byte.MaxValue;    //255
+            public static decimal CURRENCYMIN = -922337203685477.5808M;
+            public static decimal CURRENCYMAX = 922337203685477.5807M;
+            public static double SINGLEMIN = float.MinValue; // -3402823E38;
+            public static double SINGLEMAX = float.MaxValue;  //3402823E38;
```

Comment:
I'm treating `double` as essentially limitless since IIRC, when I attempted to enter a number greater than the max value for a `double`, the VBE intervenes.